commit 7ea9deb34fc56120754b29560944364e24aba62a
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sat Apr 27 15:11:18 2019 +0200

    ahci: Ignore multifunction bit in PCI header type
    
    * linux/dev/drivers/block/ahci.c (ahci_probe_dev): Clear bit 7 of
    hdrtype.

diff --git a/linux/dev/drivers/block/ahci.c b/linux/dev/drivers/block/ahci.c
index b60f1a19..3cb003c0 100644
--- a/linux/dev/drivers/block/ahci.c
+++ b/linux/dev/drivers/block/ahci.c
@@ -829,6 +829,8 @@ static void ahci_probe_dev(unsigned char bus, unsigned char device)
 		printk("ahci: %02x:%02x.%x: Can not read configuration", bus, dev, fun);
 		return;
 	}
+	/* Ignore multifunction bit */
+	hdrtype &= ~0x80;
 
 	if (hdrtype != 0) {
 		printk("ahci: %02x:%02x.%x: Unknown hdrtype %d\n", bus, dev, fun, hdrtype);
